docs/proposed/http-storage-node-protocol.rst

```diff
@@ -0,0 +1,506 @@
+
+The protocol primarily enables interaction with "resources" of two types:
+storage indexes
+and shares.
+A particular resource is addressed by the HTTP request path.
+Details about the interface are encoded in the HTTP message body.
+
+Message Encoding
+~~~~~~~~~~~~~~~~
+
+The preferred encoding for HTTP message bodies is `CBOR`_.
+A request may be submitted using an alternate encoding by declaring this in the ``Content-Type`` header.
+A request may indicate its preference for an alternate encoding in the response using the ``Accept`` header.
+These two headers are used in the typical way for an HTTP application.
+
+The only other encoding support for which is currently recommended is JSON.
+For HTTP messages carrying binary share data,
+this is expected to be a particularly poor encoding.
+However,
+for HTTP messages carrying small payloads of strings, numbers, and containers
+it is expected that JSON will be more convenient than CBOR for ad hoc testing and manual interaction.
+
+For this same reason,
+JSON is used throughout for the examples presented here.
+Because of the simple types used throughout
+and the equivalence described in `RFC 7049`_
+these examples should be representative regardless of which of these two encodings is chosen.
+
+General
+~~~~~~~
+
+``GET /v1/version``
+!!!!!!!!!!!!!!!!!!!
+
+Retrieve information about the version of the storage server.
+Information is returned as an encoded mapping.
+For example::
+
+  { "http://allmydata.org/tahoe/protocols/storage/v1" :
+    { "maximum-immutable-share-size": 1234,
+      "maximum-mutable-share-size": 1235,
+      "available-space": 123456,
+      "tolerates-immutable-read-overrun": true,
+      "delete-mutable-shares-with-zero-length-writev": true,
+      "fills-holes-with-zero-bytes": true,
+      "prevents-read-past-end-of-share-data": true,
+      "http-protocol-available": true
+      },
+    "application-version": "1.13.0"
+    }
+
+Immutable
+---------
+
+Writing
+~~~~~~~
+
+``POST /v1/immutable/:storage_index``
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+Initialize an immutable storage index with some buckets.
+The buckets may have share data written to them once.
+Details of the buckets to create are encoded in the request body.
+For example::
+
+  {"renew-secret": "efgh", "cancel-secret": "ijkl",
+   "share-numbers": [1, 7, ...], "allocated-size": 12345}
+
+The response body includes encoded information about the created buckets.
+For example::
+
+  {"already-have": [1, ...], "allocated": [7, ...]}
+
+Discussion
+``````````
+
+We considered making this ``POST /v1/immutable`` instead.
+The motivation was to keep *storage index* out of the request URL.
+Request URLs have an elevated chance of being logged by something.
+We were concerned that having the *storage index* logged may increase some risks.
+However, we decided this does not matter because the *storage index* can only be used to read the share (which is ciphertext).
+TODO Verify this conclusion.
+
```

Comment:
Sounds valid to me, but also yes lets verify :)
(Basically: does the storage-index need to be treated as sensitive information, right?)